ingestion/src/metadata/ingestion/ometa/ometa_api.py

```diff
@@ -142,30 +138,18 @@ class OpenMetadata(
 
     def __init__(self, config: OpenMetadataServerConfig, raw_data: bool = False):
         self.config = config
-        if self.config.authProvider.value == "google":
-            self._auth_provider: AuthenticationProvider = (
-                GoogleAuthenticationProvider.create(self.config)
-            )
-        elif self.config.authProvider.value == "okta":
-            self._auth_provider: AuthenticationProvider = (
-                OktaAuthenticationProvider.create(self.config)
-            )
-        elif self.config.authProvider.value == "auth0":
-            self._auth_provider: AuthenticationProvider = (
-                Auth0AuthenticationProvider.create(self.config)
-            )
-        elif self.config.authProvider.value == "azure":
-            self._auth_provider: AuthenticationProvider = (
-                AzureAuthenticationProvider.create(self.config)
-            )
-        elif self.config.authProvider.value == "custom-oidc":
-            self._auth_provider: AuthenticationProvider = (
-                CustomOIDCAuthenticationProvider.create(self.config)
-            )
-        else:
-            self._auth_provider: AuthenticationProvider = (
-                NoOpAuthenticationProvider.create(self.config)
+
+        # Load the auth provider init from the registry
+        auth_provider_fn = auth_provider_registry.registry.get(
+            self.config.authProvider.value
+        )
+        if not auth_provider_fn:
+            raise InvalidAuthProviderException(
+                f"Cannot find {self.config.authProvider.value} in {auth_provider_registry.registry}"
             )
+
+        self._auth_provider = auth_provider_fn(self.config)
```

Comment:
No need to keep updating ometa init for new providers, just update the registry after creating the class and generating the JSON Schema